File: leetcode/exercises/9.medium.longest_consecutive_sequence/longest_consecutive_sequence_1.js
// -----------------------
// Created    : 08/11/2025
// Last Edited: 08/11/2025 
// Topics     : 
// Big O      :
// Problem Id : 238
// -----------------------

/*
* @param {number[]} nums
* @return {number}
*/
var longestConsecutive = function(nums) {
        const unique_nums = new Set(nums)

        let max_seq_size = 0
        let curr_seq_size;
        for (const num of unique_nums) {
            if (!unique_nums.has(num-1)) {
                curr_seq_size = 1 
                while (unique_nums.has(num+curr_seq_size)) {
                    curr_seq_size += 1
                }
                max_seq_size = Math.max(max_seq_size, curr_seq_size)
            }
        }

        return max_seq_size
}

console.log(longestConsecutive([100,4,200,1,3,2]))

File: leetcode/exercises/9.medium.longest_consecutive_sequence/longest_consecutive_sequence_1.go
// -----------------------
// Created    : 08/11/2025
// Last Edited: 08/11/2025 
// Topics     : 
// Big O      :
// Problem Id : 238
// -----------------------

package main
import (
    "fmt"
)

func longestConsecutive(nums []int) int {

    set := make(map[int]struct{})
    for _, i := range nums {
        set[i] = struct{}{}
    }

    longest := 0
    size := 0
    for v, _ := range set {
        if _, ok := set[v-1]; !ok {
            size = 1
            for {
                if _, ok := set[v+size]; !ok {
                    break
                }
                size++
            }
        }
        longest = max(longest, size)
    }

    return longest

}

func main() {
    fmt.Print(longestConsecutive([]int{100,4,200,1,3,2}))
}
